versioned/storage/cassandra/src/main/java/org/projectnessie/versioned/storage/cassandra/CassandraBackend.java

```diff
@@ -329,16 +334,40 @@ boolean executeCas(BoundStatement stmt) {
   }
 
   ResultSet execute(BoundStatement stmt) {
-    return session.execute(stmt);
+    try {
+      return session.execute(stmt);
+    } catch (DriverException e) {
+      throw unhandledException(e);
+    }
   }
 
   CompletionStage<AsyncResultSet> executeAsync(BoundStatement stmt) {
     return session.executeAsync(stmt);
   }
 
-  void handleDriverException(DriverException e) {
-    if (e instanceof CASWriteUnknownException) {
-      logCASWriteUnknown((CASWriteUnknownException) e);
+  static RuntimeException unhandledException(DriverException e) {
+    if (e instanceof QueryExecutionException) {
+      return new UnknownOperationResultException(e);
+    } else if (e instanceof DriverTimeoutException
+        || e instanceof NodeUnavailableException
```

Comment:
`NodeUnavailable` is safe imho, this means that we couldn't even connect it, let alone send a query to it.

Reply:
However the javadoc says: "A common reason to encounter this error is when the configured number of connections per node and requests per connection is not high enough to absorb the overall request rate."

Reply:
Right; the only place where this error is raised is when there is no Netty channel available in the pool:

https://github.com/apache/cassandra-java-driver/blob/16260261d3df50fcf24fac1fc2d37896c4a111bf/core/src/main/java/com/datastax/oss/driver/internal/core/cql/CqlRequestHandler.java#L255-L259

So yes, I think we can safely assume the coordinator never got the write.